### PR TITLE
Add findChildren

### PR DESCRIPTION
**Pull request by sisp**
An attempt at closing #247.

### CODE REVIEW

File: packages/lib/src/parent/findChildren.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * given predicate matches.
+ *
+ * @param root Root object to get the matching children from.
+ * @param predicate Function that will be run for every child of the root object.
+ * @returns A readonly observable set with the matching children.
+ */
+export function findChildren<T extends object = any>(
+  root: object,
+  predicate: (node: object) => boolean
+): ReadonlySet<T> {
```
Comment (xaviergonz) [inline packages/lib/src/parent/findChildren.ts:14]: maybe add
```
,
  options?: {
    deep?: boolean
  }
```
?

**Review by xaviergonz — COMMENTED** [review]
hey, thanks for the PR! just left a comment there

File: packages/lib/src/parent/findChildren.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  const children = getChildrenObjects(root, { deep: true })
+
+  const set = new Set<any>()
```
Comment (xaviergonz) [inline packages/lib/src/parent/findChildren.ts:17]: I think it can be Set<object>

Comment (sisp) [reply packages/lib/src/parent/findChildren.ts:17]: Then I get:
```
Type 'Set<object>' is not assignable to type 'ReadonlySet<T>'.
  Types of property 'forEach' are incompatible.
    Type '(callbackfn: (value: object, value2: object, set: Set<object>) => void, thisArg?: any) => void' is not assignable to type '(callbackfn: (value: T, value2: T, set: ReadonlySet<T>) => void, thisArg?: any) => void'.
      Types of parameters 'callbackfn' and 'callbackfn' are incompatible.
        Types of parameters 'value' and 'value' are incompatible.
          Type 'object' is not assignable to type 'T'.
            'object' is assignable to the constraint of type 'T', but 'T' could be instantiated with a different subtype of constraint 'object'.ts(2322)
```

**Comment by xaviergonz** [comment]
Thanks!

**Pull request merged**